### PR TITLE
resolves #4 - Use Geopy to compute distances

### LINKED ISSUE #4: Use Geopy to compute distances
Shapely does not support coordinate system transformations. All operations on two or more features presume that the features exist in the same Cartesian plane.
https://shapely.readthedocs.io/en/stable/manual.html#coordinate-systems

Geopy can calculate geodesic distance between two points using the geodesic distance or the great-circle distance, with a default of the geodesic distance available as the function geopy.distance.distance.
https://geopy.readthedocs.io/en/stable/#module-geopy.distance

### PR DESCRIPTION
**Pull request by git-it**
This pull request resolves #4 

Added a function _measure_distance_geodesic_ to geometry_utils. 
This method replaces _geometry_utils.measure_distance_euclidean_

Tested on windows in a fresh conda environment.

Hope this helps.

### CODE REVIEW

File: movingpandas/geometry_utils.py [hunk 1]
```diff
@@ -34,6 +35,19 @@ def measure_distance_euclidean(point1, point2):
         raise TypeError("Only Points are supported as arguments, got {} and {}".format(point1, point2))
     return point1.distance(point2)
 
+def measure_distance_geodesic(point1, point2):
+    """
+    Return the geodesic distance between two points as a float of (SI) unit meters.
```
Comment (anitagraser) [inline movingpandas/geometry_utils.py:40]: This function assumes, lat/lon points, right?

File: movingpandas/geometry_utils.py [hunk 2]
```diff
@@ -42,7 +56,7 @@ def _measure_distance(point1, point2, spherical=False):
     if spherical:
         return measure_distance_spherical(point1, point2)
     else:
-        return measure_distance_euclidean(point1, point2)
+        return measure_distance_geodesic(point1, point2)
```
Comment (anitagraser) [inline movingpandas/geometry_utils.py:59]: Geodesic distance should replace spherical distance. Euclidean distance should remain the default if the CRS is not spherical

File: movingpandas/tests/test_geometry_utils.py [hunk 3]
```diff
@@ -64,4 +64,5 @@ def test_anglular_difference_twonegative(self):
         assert angular_difference(-200, -160) == 40
 
     def test_mrr_diagonal(self):
-        assert mrr_diagonal([Point(0, 0), Point(0, 2), Point(2, 0), Point(2, 2)]) == sqrt(8)
+        """Distance between NYC, NY USA and Los Angeles, CA USA is 3944411.0951634306 meters"""
```
Comment (anitagraser) [inline movingpandas/tests/test_geometry_utils.py:67]: Please create a dedicated test for the geodesic distance function and revert the change to this test.

File: movingpandas/geometry_utils.py [hunk 4]
```diff
@@ -34,6 +35,19 @@ def measure_distance_euclidean(point1, point2):
...
+        raise TypeError("Only Points are supported as arguments, got {} and {}".format(point1, point2))
+    lon1 = float(point1.x)
+    lon2 = float(point2.x)
+    lat1 = float(point1.y)
+    lat2 = float(point2.y)
+    dist = distance.distance((lat1,lon1), (lat2,lon2)).meters #uses geodesic dist and defaults to WGS84
+    return dist
+
 
 def _measure_distance(point1, point2, spherical=False):
```
Comment (anitagraser) [inline movingpandas/geometry_utils.py:52]: The docstring will need to be updated to reflect changes. There are probably other docstrings that are related to this as well.

**Review by anitagraser — CHANGES_REQUESTED** [review]

File: movingpandas/geometry_utils.py (hunk 1 above)
Comment (git-it) [reply movingpandas/geometry_utils.py:40]: Yes, the function inputs (Shapely.geometry) Point objects. The x and y are assumed to be latitude / longitude.
In addition, it should be noted that the method ```distance.distance((lat1,lon1), (lat2,lon2))``` defaults to the WGS-84 datum. WGS-84 is used by GPS and is likely the most common datum.

The datum can be changed like so ```distance.geodesic(point1, point2, ellipsoid='GRS-67').meters)```

File: movingpandas/geometry_utils.py [hunk 5]
```diff
@@ -34,13 +35,33 @@ def measure_distance_euclidean(point1, point2):
...
+def measure_distance_geodesic(point1, point2):
+    """
+    This function calculates the geodesic distance between two points 
+    as a float of (SI) unit meters.
+
+    inputs:
+    * point1 - a shapely.geometry.Point object containing latitude / longitude defining a starting location
+    * point1 - a shapely.geometry.Point object containing latitude / longitude defining a termination location
+    Output:
+    * dist - a float describing the geodesic distance (on a WGS84 ellipsoid) in meters
```
Comment (anitagraser) [inline movingpandas/geometry_utils.py:47]:
```suggestion
    Parameters
    ----------
    point1 : shapely.geometry.Point 
        point object containing latitude / longitude defining a starting location
    point2 : shapely.geometry.Point 
        object containing latitude / longitude defining a termination location
        
    Returns
    -------
    dist : float 
        geodesic distance (on a WGS84 ellipsoid) in meters
```

**Review by anitagraser — APPROVED** [review]

**Comment by anitagraser** [comment]
Thank you for your contribution! 

I'm not sure why Codecov thinks the geodesic distance function is not covered by tests but everything else looks good now.
